### DISCUSSION TITLE
如何在Markdown内插入图片及调整（相对）大小？

### DISCUSSION THREAD

**Discussion by leimao** [ppoffice/hexo-theme-icarus#988] [Q&A]
看了一下似乎没有教程。hexo的官方图片插入方法似乎也不起作用。谢谢。

**Comment by leimao** [comment]
官方指导只适用于调整绝对长度：
https://ppoffice.github.io/hexo-theme-icarus/uncategorized/hexo-built-in-tag-helpers/#Image

另外，似乎图片在HTML显示的title是alt而不是title？

**Reply by Xiang3999** [reply]
请问这个问题你解决了吗？怎么插入照片？

**Reply by leimao** [reply]
@Xiang3999 目前为止没有，我使用的还是绝对长度。

**Comment by realzza** [comment]
leimao大佬好！

我个人用下来，除了正常的markdown内插入图片之外，也能直接用html在markdown里插入图片
例如：
```html
<center><img src="../../assets/img/sample.JPG" alt="jpg name" width="60%"/></center>
```
`<center>`是为了让图片居中，`width`可以调整图片占比，`alt`是图片title
唯一需要注意的是，`src`地址的参考系在hexo编译之后的`public`中，可能与`source`中文件的hierarchy不同。

**Reply by leimao** [reply]
@realzza 基本所有博客的Markdown都支持直接插入HTML。缺点是没有网站的CSS可以给图片修饰。比如说你想要原来网站自带的圆润的图片边角效果，你得研究网站源代码，找出对应的CSS，然后对插入的HTML进行修改，比较麻烦。

**Reply by realzza** [reply]
@leimao 是的，需要看看源码，但也有相对简单的解决办法。我理解只需要在对应`styl`文件中作修改实现圆角这些css效果。`styl`本身也就是编译前的css的作用。统一设置文章中的图片格式的话，在https://github.com/ppoffice/hexo-theme-icarus/blob/30efe838c5eaa8885da1e33e986f9b1ea1866802/include/style/article.styl#L53 里面加一个`img`的定义就好了。圆角效果可以加上，编译后会体现在css中。
```styl
.content
    img 
        border-radius: 5px
        margin-bottom: 10px 
        box-shadow: 0 0 10px rgba(0, 0, 0, 0.2)
```
需要自定义图片宽度的时候，再通过html里面改width就好
